grand/simulation/antenna/generic.py

```diff
@@ -139,22 +155,41 @@ def fftfreq(n, t):
         if dir_frame is not None:
             # Change the direction to the antenna frame
             if isinstance(direction, BaseRepresentation):
-                direction = dir_frame.realize_frame(direction)
-            direction = direction.transform_to(antenna_frame) # Now compute direction vector data in antenna frame
-            direction = direction.data
+                print("dir pre", direction, dir_frame, antenna_frame)
+                # LWP: this joins direction and dir_frame - not needed if we get rid of astropy
+                if grand_astropy:
+                    direction = dir_frame.realize_frame(direction)
+                else:
+                    E = field.E
+                
+                print("dir post", direction)
+                #exit()
+            
+            # LWP: This adds the difference of the bases (shower_frame-antenna_frame) to the direction, which can be done manually
+            if grand_astropy:
+                direction = direction.transform_to(antenna_frame).data
+                print("dire postpost", direction)
+            else:
```

Comment:
Same as comment above. In this case the direction should be rotated, not translated.